liblouis/lou_translateString.c

```diff
@@ -1992,240 +1992,249 @@ for_selectRule(const TranslationTableHeader *table, int pos, OutString output, i
 							(beforeAttributes & (*transRule)->after)) &&
 						(!((*transRule)->before & ~CTC_EmpMatch) ||
 								(afterAttributes & (*transRule)->before)))
-					switch (*transOpcode) { /* check validity of this Translation */
-					case CTO_Space:
-					case CTO_Letter:
-					case CTO_UpperCase:
-					case CTO_LowerCase:
-					case CTO_Digit:
-					case CTO_LitDigit:
-					case CTO_Punctuation:
-					case CTO_Math:
-					case CTO_Sign:
-					case CTO_Hyphen:
-					case CTO_Replace:
-					case CTO_CompBrl:
-					case CTO_Literal:
-						return;
-					case CTO_Repeated:
-						if (dontContract || (mode & noContractions)) break;
-						if ((mode & (compbrlAtCursor | compbrlLeftCursor)) &&
-								pos >= compbrlStart && pos <= compbrlEnd)
+					/* check nocross */
+					if (!((*transRule)->nocross &&
```

Comment:
This diff is really hard to read (at least in gh). The way I read it is that liblouis will only switch on the `transOpcode` if we are not looking at a nocross with a hyphenation table. What happens in the other cases? From what I can see nothing seems to happen. I don't see an else branch.

Reply:
Yeah I know, these kind of diffs suck, nothing I can do about that (except for making a format fix in a separate commit). But nothing changed much. I basically just added this if-then. If the condition is not matched, we just skip the rule and check the next one.